haystack/components/evaluators/faithfulness.py

```diff
@@ -0,0 +1,170 @@
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+
+class FaithfulnessEvaluator(LLMEvaluator):
+    """
+    Evaluator that checks if a generated answer can be inferred from the provided contexts.
+
+    An LLM separates the answer into multiple statements and checks whether the statement can be inferred from the
+    context or not. The final score for the full answer is a number from 0.0 to 1.0. It represents the proportion of
+    statements that can be inferred from the provided contexts.
+
+    Usage example:
+    ```python
+    from haystack.components.evaluators import FaithfulnessEvaluator
+
+    questions = ["Which is the most popular global sport?", "Who created the Python language?"]
+    contexts = [
+        [
+            "The popularity of sports can be measured in various ways, including TV viewership, social media presence, number of participants, and economic impact. Football is undoubtedly the world's most popular sport with major events like the FIFA World Cup and sports personalities like Ronaldo and Messi, drawing a followership of more than 4 billion people."
+        ],
+        [
+            "Python, created by Guido van Rossum in the late 1980s, is a high-level general-purpose programming language. Its design philosophy emphasizes code readability, and its language constructs aim to help programmers write clear, logical code for both small and large-scale software projects."
+        ],
+    ]
+    responses = [
+        "Football is the most popular sport with around 4 billion followers worldwide.",
+        "Python is a high-level general-purpose programming language that was created by George Lucas.",
+    ]
+    evaluator = FaithfulnessEvaluator()
+    result = evaluator.run(
+        questions=questions,
+        contexts=contexts,
+        responses=responses,
+    )
+    print(results["evaluator"])
+    # {'results': [{'statements': ["Football is undoubtedly the world's most popular sport.", 'Football has around 4
+    # billion followers worldwide.'], 'statement_scores': [1, 1], 'name': 'llm', 'score': 1.0}, {'statements': ['Python
+    # is a high-level general-purpose programming language.', 'Python was created by George Lucas.'], 'statement_scores':
+    # [1, 0], 'name': 'llm', 'score': 0.5}], 'score': 0.75, 'individual_scores': [1.0, 0.5]}
+
+    ```
+    """
+
+    def __init__(
+        self,
+        instructions: str = "Your task is to judge the faithfulness or groundedness of statements based on context "
+        "information. First, please extract statements from a provided response to a question. "
+        "Second, calculate a faithfulness score for each statement made in the response. "
+        "The score is 1 if the statement can be inferred from the provided context or 0 if it "
+        "cannot be inferred.",
+        inputs: Optional[List[Tuple[str, Type[List]]]] = None,
+        outputs: Optional[List[str]] = None,
+        examples: Optional[List[Dict[str, Any]]] = None,
```

Comment:
I don't think we should expose these to the user as they essentially make this component the same as the underlying `LLMEvaluator`. The purpose of this component is to provide a plug-and-play metric for the user that has a predefined prompt, inputs, outputs and examples. 

If we allow the user the customize the prompt, for instance, we can no longer be certain that the semantics of the metric holds. Same with the input. And the outputs need to be pre-defined as we need to know the keys to calculate the mean and individual scores.

Reply:
Okay, fair point. The examples we still want to expose right?

Reply:
We could rename them to `additional_examples` and then document one of the pre-defined examples in the parameters docstring.